### ISSUE TITLE
NX cache do not work with runExecutor ( composing executors )

### ISSUE THREAD

**Issue by radswiat** [nrwl/nx#13579]
### Current Behavior

When running executor from terminal, executor cache works fine:

```
nx run projectA:genVersion
```

When using the same executor from combined command, cache doesnt work:

```
nx run projectA:build
```

```ts
export default async function buildComposedExecutor(options: StartInt1ExecutorSchema, context: ExecutorContext) {
  await runExecutor({ project: 'projectA', target: 'clean' }, options, context)
  await runExecutor({ project: 'projectA', target: 'genVersion' }, options, context)
  return {
    success: true,
  }
}
```

### Expected Behavior

Executor cache should work when used within other executor.

### Steps to Reproduce

Create 2 custom executors:
1. Standalone one with a cache configuration
2. Composed one that uses the first executor
3. Run first standalone to check that cache is working
4. Run second and expect the cache to work (broken)

### Nx Report

```shell
>  NX   Report complete - copy this into the issue template

   Node : 14.19.0
   OS   : darwin x64
   yarn : 1.22.19
   
   nx : 15.0.13
   @nrwl/angular : Not Found
   @nrwl/cypress : Not Found
   @nrwl/detox : Not Found
   @nrwl/devkit : 15.2.4
   @nrwl/esbuild : Not Found
   @nrwl/eslint-plugin-nx : Not Found
   @nrwl/expo : Not Found
   @nrwl/express : Not Found
   @nrwl/jest : 15.2.4
   @nrwl/js : 15.2.4
   @nrwl/linter : 15.2.4
   @nrwl/nest : Not Found
   @nrwl/next : Not Found
   @nrwl/node : Not Found
   @nrwl/nx-cloud : Not Found
   @nrwl/nx-plugin : 15.2.4
   @nrwl/react : Not Found
   @nrwl/react-native : Not Found
   @nrwl/rollup : Not Found
   @nrwl/schematics : Not Found
   @nrwl/storybook : Not Found
   @nrwl/web : Not Found
   @nrwl/webpack : Not Found
   @nrwl/workspace : 15.0.13
   typescript : 4.1.5
   ---------------------------------------
   Local workspace plugins:
         @web-xxx/xxx-scripts
   ---------------------------------------
   Community plugins:
```

### Additional Information

There is no documentation anywhere about cache for composed executors - so I'm not sure if this supposed to work that way, but I think it kinda makes sense as otherwise how we could take benefit of cache for complex executors?

Thanks :)

**Comment by woppa684** [comment]
My guess is that both of the executors run on the same output folder? What happens if you don't use `runExecotor` but instead directly call the executor functions?

**Comment by radswiat** [comment]
I've tried all 3 options:
- runExecotor (no cache)
- import executor and call directly (no cache)
- run from command line using nx run (cache works)

I have 3 tasks:
- clean (no cache)
- generateVersion (cached)
- start (composed) (no cache)

As you can see there is no cache configuration conflict as only one executor have cache enabled.

One of the problems with debugging it is that when running as function or runExecutor there is no information given about the cache used for each of the commands.

Basically what I need is for each executor to have its own independent cache as i'm building the app using multiple commands and some will be cached, some will not. And caching will vary depending on the command.

I could create my own executor calling `nx run` which might fix the issue - but wondering if this shouldn't be built in.

Thanks :)

**Comment by patdx** [comment]
> Basically what I need is for each executor to have its own independent cache as i'm building the app using multiple commands and some will be cached, some will not. And caching will vary depending on the command.

I was thinking about a use case which might be similar. I want to run an uncached command after a cached command as part of the `build` target. Not 100% sure if related or not, but I've created a discussion topic for that. Just in case they could be related, here is the link: https://github.com/nrwl/nx/discussions/14541
